pkgs/development/haskell-modules/hackage-packages.nix

```diff
@@ -185893,7 +185894,7 @@ self: {
          base containers doctest ghc ghc-tcplugins-extra hspec
          inspection-testing polysemy should-not-typecheck syb transformers
        ];
-       testToolDepends = [ hspec-discover ];
+       testToolDepends = [ hspec-discover cabal-doctest ];
```

Comment:
You're correct that this file is autogenerated, so it shouldn't be edited by hand.

Above I suggest just using `dontCheck` for polysemy-plugin, but in case you're curious, you can grep through `pkgs/development/haskell-modules/configuration-common.nix` to see how people add test tools.

One solution is to use the `addBuildTool` function.  The other solution is to use `overrideCabal` and explicitly set `testToolDepends`.

Reply:
`cabal-doctest` should probably go into `setupHaskellDepends` since it's a part of the `Setup.hs` build. This means that this dependency will be required regardless of whether testing is enabled or disabled.